Source/ACE.Server/WorldObjects/Portal.cs

```diff
@@ -142,53 +142,53 @@ public override ActivationResult CheckUseRequirements(WorldObject activator)
                 else if (PropertyManager.GetBool("pkl_server").Item)
                     playerPkLevel = PKLevel.PKLite;
 
-                if ((PortalRestrictions & PortalBitmask.Undef) != 0)
+                if (PortalRestrictions == PortalBitmask.NotPassable)
                 {
                     // Players may not interact with that portal.
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PlayersMayNotUsePortal));
                 }
 
-                if ((PortalRestrictions & PortalBitmask.NoPk) != 0 && (playerPkLevel == PKLevel.PK || player.PlayerKillerStatus == PlayerKillerStatus.PK))
+                if (PortalRestrictions.HasFlag(PortalBitmask.NoPk) && (playerPkLevel == PKLevel.PK || player.PlayerKillerStatus == PlayerKillerStatus.PK))
                 {
                     // Player killers may not interact with that portal!
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PKsMayNotUsePortal));
                 }
 
-                if ((PortalRestrictions & PortalBitmask.NoPKLite) != 0 && (playerPkLevel == PKLevel.PKLite || player.PlayerKillerStatus == PlayerKillerStatus.PKLite))
+                if (PortalRestrictions.HasFlag(PortalBitmask.NoPKLite) && (playerPkLevel == PKLevel.PKLite || player.PlayerKillerStatus == PlayerKillerStatus.PKLite))
                 {
                     // Lite Player Killers may not interact with that portal!
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.PKLiteMayNotUsePortal));
                 }
 
-                if ((PortalRestrictions & PortalBitmask.NoNPK) != 0 && playerPkLevel == PKLevel.NPK)
+                if (PortalRestrictions.HasFlag(PortalBitmask.NoNPK) && playerPkLevel == PKLevel.NPK)
                 {
                     // Non-player killers may not interact with that portal!
                     return new ActivationResult(new GameEventWeenieError(player.Session, WeenieError.NonPKsMayNotUsePortal));
                 }
 
-                if ((PortalRestrictions & PortalBitmask.OnlyOlthoiPCs) != 0
-                    && player.WeenieClassId != 43480 && player.WeenieClassId != 43481
-                    && player.WeenieClassId != 43493 && player.WeenieClassId != 43494)
+                if (PortalRestrictions.HasFlag(PortalBitmask.OnlyOlthoiPCs)
+                    && !player.WeenieClassName.Equals("olthoiplayer", System.StringComparison.OrdinalIgnoreCase) && !player.WeenieClassName.Equals("olthoiacidplayer", System.StringComparison.OrdinalIgnoreCase)
```

Comment:
er not exactly what i mean...

string search is much slower than comparing wcids

are these not available in WeenieClassName enum? if not, just comment them